src/graph/heterograph.cc

```diff
@@ -277,6 +277,24 @@ HeteroGraphPtr HeteroGraph::CopyTo(HeteroGraphPtr g, const DGLContext& ctx) {
       hgindex->meta_graph_, rel_graphs, hgindex->num_verts_per_type_));
 }
 
+HeteroGraphPtr HeteroGraph::PinMemory(HeteroGraphPtr g) {
+  bool allpinned = true;
+  std::vector<HeteroGraphPtr> rel_graphs;
+  auto hgindex = std::dynamic_pointer_cast<HeteroGraph>(g);
+  CHECK_NOTNULL(hgindex);
+  for (auto subg : hgindex->relation_graphs_) {
```

Comment:
I'd prefer to have 2 loops here. 
The first just check whether all pined.
If not, loop again and push.
The loop should be way cheaper than push_back ops.